### PR TITLE
Update `main` post-23.06 release

### PR DESCRIPTION
**Pull request by mc-nv**

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -1189,7 +1189,55 @@ def dockerfile_prepare_container_linux(argmap, backends, enable_gpu,
  && rm -f ${_CUDA_COMPAT_PATH}/lib
 '''
     else:
-        add_cpu_libs_to_linux_dockerfile(df, backends, target_machine)
+        libs_arch = 'aarch64' if target_machine == 'aarch64' else 'x86_64'
```
Comment (dyastremsky) [inline build.py:1192]: Thanks for doing the hard job of updating main based on the 23.06 commits. This code chunk isn't needed though. The line that these commit seem to be deleting right above (add_cpu_libs_to_linux_dockerfile) was refactored [here](https://github.com/triton-inference-server/server/pull/5990), which is why it's no longer in server. So the current code, without these additions (and the removal of add_cpu_libs_to_linux dockerfile above), is correct.

**Review by dyastremsky — CHANGES_REQUESTED** [review]
PR looks good to me aside from needing to revert the changes to the else block in build.py's `dockerfile_prepare_container_linux` function.

File: build.py (hunk 1 above)
Comment (mc-nv) [reply build.py]: I have dropped that commit for the PR.

**Review by Tabrizian — DISMISSED** [review]

File: build.py (hunk 1 above)
Comment (dyastremsky) [reply build.py]: Thanks, looks good!

**Review by dyastremsky — DISMISSED** [review]

**Review by Tabrizian — APPROVED** [review]

**Pull request merged**